### PR TITLE
CLOUDP-156416: Add Atlas CLI endpoint for programmatic org creation

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-156416

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments
Manually tested and e2e tested

when using API keys
```bash
./bin/atlas orgs create Test --ownerId <>  --apiKeyRole ORG_OWNER --apiKeyDescription "test"
```

When using atlas login
```bash
./bin/atlas orgs create Test
```

### CODE REVIEW

File: internal/cli/iam/organizations/create_atlas.go
Comment (sarahsimpers) [inline internal/cli/iam/organizations/create_atlas.go:127]: Question: Can any role this command? If not, can we add the required role as Long text here (replacing Organization Project Creator with whatever the required role is)?
```suggestion
		Long: fmt.Sprintf(usage.RequiredRole, "Organization Project Creator"),
		Args:  require.ExactArgs(1),
```

**Review by sarahsimpers — APPROVED** [review]
LGTM with one question/suggestion if it's appropriate

File: internal/cli/iam/organizations/create_atlas.go
Comment (gssbzn) [reply internal/cli/iam/organizations/create_atlas.go:127]: Complicated, this command has two behaviors, either using API keys, it will need to be org owner and belong to an org with cross org billing enabled, or when using `atlas login` in which case any role

Comment (sarahsimpers) [reply internal/cli/iam/organizations/create_atlas.go:127]: Ah ok, let's hold off on the above suggestion then. Thanks!

**Review by wtrocki — APPROVED** [review]
LGTM

**Pull request merged**
